android/src/main/java/com/oney/WebRTCModule/WebRTCView.java

```diff
@@ -418,21 +187,12 @@ public void setObjectFit(String objectFit) {
     }
 
     private void setScalingType(ScalingType scalingType) {
-        SurfaceViewRenderer surfaceViewRenderer;
-
-        synchronized (layoutSyncRoot) {
-            if (this.scalingType == scalingType) {
-                return;
-            }
-
+        if (this.scalingType != scalingType) {
             this.scalingType = scalingType;
 
             surfaceViewRenderer = getSurfaceViewRenderer();
```

Comment:
Assigning to `surfaceViewRenderer` causes a compile-time error because the field is final. You didn't really want to assign to the field but it merely happened as a side effect of deleting `SurfaceViewRenderer surfaceViewRenderer;` at the beginning of the function. That was easy to fix because a local variable wasn't really anyway and it was OK to simply write `getSurfaceViewRenderer().setScalingType(scalingType)`. Anyway, I hit another compile-type error so I'm not amending your commit right away.